tsl/src/hypertable.c

```diff
@@ -164,7 +165,12 @@ hypertable_get_and_validate_data_nodes(ArrayType *nodearr)
 		ereport(ERROR,
 				(errcode(ERRCODE_TS_INSUFFICIENT_NUM_DATA_NODES),
 				 errmsg("no data nodes can be assigned to the hypertable"),
-				 errhint("Add data nodes to the database.")));
+				 errdetail(nodearr == NULL && list_length(all_data_nodes) == 0 ?
```

Comment:
Small nitpick, in the code there are 3 consecutive uses of `list_length(all_data_nodes)` and 2 consecutive uses of `(nodearr == NULL && list_length(all_data_nodes) == 0)`

It would be easier to read if it was something like:
```
int num_all_data_nodes = list_length(all_data_nodes);
bool permission_error = !(nodearr == NULL && num_all_data_nodes == 0);
```

The same thing happens in line 152 where we use `list_length(data_nodes);` when we already have `num_data_nodes` set. We could fix that as well.

Reply:
I do not think that `num_all_data_nodes` is easier to read than `list_length(all_data_nodes)`, they are pretty much equivalent.